src/main/java/com/minecolonies/coremod/commands/ColonyTPCommand.java

```diff
@@ -0,0 +1,236 @@
+     *
+     * @param blockPos for the current block LOC
+     * @return blockPos to be used for the TP
+     */
+    private static BlockPos findLand(final BlockPos blockPos, final World world)
+    {
+        int top = STARTING_Y;
+        int bot = 0;
+        int mid = STARTING_Y;
+
+        BlockPos foundland = null;
+        BlockPos tempPos = blockPos;
+        //We are doing a binary search to limit the amount of checks (usually at most 9 this way)
+        while (top >= bot)
+        {
+            tempPos = new BlockPos( tempPos.getX(),mid, tempPos.getZ());
+            Block blocks = world.getBlockState(tempPos).getBlock();
+            if (blocks == Blocks.AIR && world.canSeeSky(tempPos))
+            {
+                top = mid - 1;
+                foundland = tempPos;
+            }
+            else
+            {
+                bot = mid + 1;
+                foundland = tempPos;
+            }
+            mid = (bot + top)/2;
+        }
+
+        return foundland;
+    }
+
+    /**
+     * this checks that you are not in liquid.  Will check for all liquids, even those from other mods
+     * before TP
+     *
+     * @param blockPos for the current block LOC
+     * @param sender uses the player to get the world
+     * @return isSafe true=safe false=water or lava
+     */
+    private static boolean isPositionSave(@NotNull ICommandSender sender, BlockPos blockPos)
+    {
+        return sender.getEntityWorld().getBlockState(blockPos).getBlock() != Blocks.AIR && !sender.getEntityWorld().getBlockState(blockPos).getMaterial().isLiquid();
+    }
+    /**
+     * this checks that you are not too close to another colony
+     * before TP
+     *
+     * @param blockPos for the current block LOC
+     * @return colNear false=no true=yes
+     */
+    private boolean findColony(BlockPos blockPos, World world)
+    {
+        Colony nearestCol = ColonyManager.getClosestColony(world, blockPos);
+        Boolean colNear;
+        /* get individual coords to do the math */
+        int cx = nearestCol != null ? nearestCol.getCenter().getX() : 0;
+        int cz = nearestCol != null ? nearestCol.getCenter().getZ() : 0;
+        /* from the random X for the TP */
+        double px = blockPos.getX();
+        /* from the random Z for the TP */
+        double pz = blockPos.getZ();
+
+        double dist = Math.sqrt(Math.pow(cx - px, 2.0) + (Math.pow(cz - pz, 2.0)));
+                        /* grab the working distance and do our check now that we have distance from nearest colony*/
+                        /* just to understand this better::::
+                            I am taking the working distance from the town hall and doubling it this will give me
+                            the distance needed from both town halls -if you placed it here.  Then im adding on 20%
+                            just to get a padding between the two.*/
+        double wd = (Configurations.workingRangeTownHall * 2) * ADDS_TWENTY_PERCENT;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Local variable 'wd' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)